apps/web/src/state/farmsV4/state/poolApr/fetcher.ts

```diff
@@ -141,3 +182,168 @@ export const getAllNetworkMerklApr = async () => {
   const aprs = await Promise.all(supportedChainIdV4.map((chainId) => getMerklApr(chainId)))
   return aprs.reduce((acc, apr) => ({ ...acc, ...apr }), {})
 }
+
+const getV3PoolsCakeAprByChainId = async (pools: V3PoolInfo[], chainId: number, cakePrice: BigNumber) => {
+  const masterChefV3 = getMasterChefV3Contract(undefined, chainId)
+  const client = publicClient({ chainId })
+  const validPools = pools.filter((pool) => {
+    return pool.pid && pool.chainId === chainId
+  })
+
+  if (!masterChefV3 || !client) return {}
+
+  const masterChefV3Cache = masterChefV3CacheMap.get(chainId)
+  const [totalAllocPoint, latestPeriodCakePerSecond] = await Promise.all([
+    masterChefV3Cache ? masterChefV3CacheMap.get(chainId)!.totalAllocPoint : masterChefV3.read.totalAllocPoint(),
```

Comment:
prevent to use Non-Null Assertion, i think here can be use `masterChefV3Cache` instead